### ISSUE TITLE
Query `(_ . (name) @firstChild)` captures every `name` node

### ISSUE THREAD

**Issue by RedCMD** [tree-sitter/tree-sitter#2869]
The query `(_ . (_) @firstChild)` will correctly capture the first child of every named node
returns: `function_definition` and the first `identifier` ✅

```
(source_file
  (function_definition
    (identifier)
    (identifier)
    (identifier)))
```

however if you specify a node name inside of `(_)`, the query will instead capture every single node of that name
regardless of if its the first named node or not
`(_ . (identifier) @everyChild)`
returns: `function_definition` and **every single** `identifier` ❌

specifying the parent name does work correctly: `(function_definition . (identifier) @firstChild)`
returns: the first `identifier` ✅

https://github.com/tree-sitter/tree-sitter/blob/master/docs/section-2-using-parsers.md#anchors
https://github.com/tree-sitter/tree-sitter/blob/master/docs/section-2-using-parsers.md#wildcard-node

**Comment by amaanq** [comment]
this isn't actionable without a grammar to go along with your query reproducer

**Comment by RedCMD** [comment]
go to https://tree-sitter.github.io/tree-sitter/playground
select `JavaScript`
paste code `function name(one, two, three) { }`
paste query `(formal_parameters . (identifier) @firstChild)`
notice how only the first `identifier` is highlighted ✅

change `formal_parameters` to `_`
`(_ . (identifier) @firstChild)`
notice how every single `identifier` is now highlighted ❌

**Comment by amaanq** [comment]
Fixed by #3440
